tests/test_snapshot.py

```diff
@@ -473,3 +478,48 @@ def test_pc_rating_rate_none(user: UserClient):
 def test_pc_2(user: UserClient):
     s = file('laptop-hp_255_g3_notebook-hewlett-packard-cnd52270fw.snapshot')
     snapshot, _ = user.post(res=Snapshot, data=s)
+
+
+@pytest.mark.mvp
+def test_save_snapshot_in_file():
+    """ This test check if works the function save_snapshot_in_file """
+    snapshot_no_hid = file('basic.snapshot.nohid')
+    save_json(snapshot_no_hid)
+
+    uuid = snapshot_no_hid['uuid']
+    files = [x for x in os.listdir(TMP_SNAPSHOTS) if uuid in x]
+
+    snapshot = {'software': '', 'version': '', 'uuid': ''}
+    if files:
+        path_snapshot = "{dir}/{file}".format(dir=TMP_SNAPSHOTS, file=files[0])
+        file_snapshot = open(path_snapshot)
+        snapshot = json.loads(file_snapshot.read())
+        file_snapshot.close()
+        os.remove(path_snapshot)
+
+    assert snapshot['software'] == snapshot_no_hid['software']
+    assert snapshot['version'] == snapshot_no_hid['version']
+    assert snapshot['uuid'] == uuid
+
+
+@pytest.mark.mvp
+def test_backup_snapshot_with_errors(user: UserClient):
+    """ This test check if the file snapshot is create when some snapshot is wrong """
+    snapshot_no_hid = file('basic.snapshot.badly_formed')
+    uuid = snapshot_no_hid['uuid']
+
+    snapshot = {'software': '', 'version': '', 'uuid': ''}
+    with pytest.raises(KeyError):
+        response = user.post(res=Snapshot, data=snapshot_no_hid)
+
+    files = [x for x in os.listdir(TMP_SNAPSHOTS) if uuid in x]
+    if files:
+        path_snapshot = "{dir}/{file}".format(dir=TMP_SNAPSHOTS, file=files[0])
+        file_snapshot = open(path_snapshot)
+        snapshot = json.loads(file_snapshot.read())
+        file_snapshot.close()
```

Comment:
Same comments as on the previous test method.

Reply:
Thank you.
I put "with statement" and I used os.path.join in the test and in the view of actions.